vignettes/examples.Rmd

```diff
@@ -12,110 +12,110 @@ vignette: >
+co2 <- add_region_ID(co2, lookupfile=rgn32, drops=rgn32)
+co2 <- dplyr::filter(co2, year==2050)
+
+### Show the first 10 observations
+co2[1:10, ]
+```
+The data is in long form, so using `dplyr::filter()`  helps to get just the observations for the year you want to plot.  At the end of this, `co2` is a data frame that has CO<sub>2</sub> emissions by region for the year 2050.  This structure can now be passed to `plot_GCAM()` to plot maps, as shown in the examples below.
+
+## Loading map data
+When using `plot_GCAM()` for your GCAM data, you also need to provide the map data that it should be associated with.  Generally you should use one of the map datasets included in the package whenever possible because they correspond to the geographical units reported by GCAM.  The datasets available are `map.rgn32`, `map.rgn14`, `map.basin235`, and `map.chn`.  They correspond to the 32-region and 14-region GCAM region maps, the 235 global water basin map, and the 32 regions plus China subregions map. Additionally, each of those datasets has a simplified version which is helpful if you need faster plotting or less cluttered maps. 
+```{r plot.default, fig.width=3.48, fig.show='hold'}
+plot_GCAM(map.rgn32, title="Full 32-Region map")
+plot_GCAM(map.rgn32.simple, title="Simplified 32-Region map")
 ```
-At the end of all this, `map_co2` is a map data frame that has primary energy consumption by region.  Each model year is a separate column.  This structure can be passed to `plot_GCAM` to plot maps.
+
+It is possible, however, for you to load your own map data as well.  The map data can be loaded by the package given the file path.  Your map data can be in any of the following formats: sf objects, spatial data frames, ESRI Shapefiles, or GeoJSON files.  You can either pass the name of the file (the full file path) to `plot_GCAM()` or load the data yourself first.
+```{r load.map, fig.width=4, fig.align='center'}
+mapdata <- system.file("extdata/rgn32", "reg32_spart.shp", package = "gcammaptools")
+mapdata
+plot_GCAM(mapdata)
+```
+This gives you the same structure you would have gotten with `data(map.rgn32)`.  Generally you should use one of the map datasets included in the package whenever possible because they correspond to the geographical units reported by GCAM.  The datasets available are `map.rgn32`, `map.rgn14`, `map.basin235`, and `map.chn`.  They correspond to the 32-region and 14-region GCAM region maps, the 235 global water basin map, and the 32 regions plus China subregions map.
 
 ## Sample Maps generated with plot_GCAM and GCAM32 shapefile
 
 ### Example 1: Eckert III World Projection, Colored by Region
 
 This example just plots the map data frame with the GCAM region id.  Thus, you get each region colored according to a discrete color palette.
 ```{r mp1, fig.width=6, fig.height=3}
-plot_GCAM(map_32_wo_Taiwan.fort, col = 'id', proj = eck3, colorfcn=qualPalette)
+plot_GCAM(mapdata, col = 'region_id', proj = eck3)
```

Comment:
This plot doesn't actually match the description above.  Instead of a discrete color palette, you get shades of blue.

Reply:
Also, is there a reason why the `region_id` columns in the maps are of type numeric rather than integer?  Having an integer type would cause R to treat it as discrete values, instead of continuous.

Reply:
I agree that `region_id` should be of type integer, but they would still be [treated as continuous values](https://github.com/tidyverse/ggplot2/issues/1340). If I add a column to the maps for region name (or add it on the fly) it can plot the regions using the `gcam32_colors` palette. Any reason not to do that?

Reply:
That sounds like the way to go.